salesagility/SuiteCRM
# Fixed #8004 - My Activity Stream Date Is Backdating

## Description



- Deprecated the ```SecurityGroupMessage``` static ```getTimeLapse``` method and the ```SugarFeed``` ```getTimeLapse``` non-static method in favour of a single function in ```TimeDate``` class. This should fix any remaining issues with activity stream dates and simplifies/unifies the TimeLapse code.
- Explicitly added PHPUnit to composer.
- Added some unit tests and fixed some inspections.

## Motivation and Context

Issue reference: #8004 

## How To Test This

I would suggest going into the database and setting a variety of different dates for something like cases and then checking that everything displays correctly on the homepage activity stream dashlet. Make sure to check that changing timezones works as expected also.

## Types of changes


### Final checklist
